pkg/machines/components/create-vm-dialog/createVmDialog.jsx

```diff
@@ -134,12 +134,7 @@ class CreateVM extends React.Component {
                     autodetectOS(os)
                             .fail(ex => console.log("osinfo-detect command failed: ", ex.message))
                             .then(res => {
-                                let osName = res.match(/OS ['"](.*)['"]/)[1];
-                                // osinfo-query is nto yet aware of variants, https://gitlab.com/libosinfo/libosinfo/issues/24
-                                // but osinfo-detect is. Remove the variant suffix to eliminate this issue
-                                osName = osName.replace(/ Server$/, "");
-                                osName = osName.replace(/ Workstation$/, "");
-                                const osEntry = this.props.osInfoList.filter(osEntry => osEntry.name == osName);
+                                const osEntry = this.props.osInfoList.filter(osEntry => osEntry.id == res.trim());
```

Comment:
Please don't repeatedly call trim() -- do it once outside the loop.

Reply:
I've also asked on IRC, but let me ask here so I can easily get back to this if needed.

Calling trim() in the value returned by autodetectOS() is something that's actually bothering me. Ideally we should trim() the return in the autodetectOS() function, not on its callers.

However, my javascript knowledge is quite limited and I didn't find an easy way to do so.
If you have any suggestion, please, let me know.

Reply:
That's a bit tricky indeed, as `autodetectOS()` really just directly returns a promise from cockpit.spawn(). A cleaner way is to use [promise chaining](https://javascript.info/promise-chaining), i. e. define `autodetectOS()` like this:

    return cockpit.spawn(...).then(out => out.trim());